### PR TITLE
Add support for yesterday's daily rewards of Hellevator

### PR DESCRIPTION
**Pull request by mzimnn**
This pull request adds support for accessing and claiming yesterday's daily rewards of the Hellevator.

As a far as I can tell,  the S&F server only returns yesterday's daily rewards during login. This is unfortunate because it requires that the game state is updated manually.

Some of the logic was already implemented but commented out. I am unsure why this was done, as it seems to work fine.

### CODE REVIEW

File: src/gamestate/unlockables.rs [hunk 1]
```diff
@@ -68,13 +68,12 @@ impl HellevatorEvent {
...
-    // /// If the Hellevator event is active, this returns a mutable reference
-    // to /// the Information about it
-    // #[must_use]
-    // pub fn active_mut(&mut self) -> Option<&mut Hellevator> {
-    //     let is_active = self.is_event_ongoing();
-    //     self.active.as_mut().filter(|_| is_active)
-    // }
+    /// If the Hellevator event is active, this returns a mutable reference
+    #[must_use]
+    pub fn active_mut(&mut self) -> Option<&mut Hellevator> {
```
Comment (the-marenga) [inline src/gamestate/unlockables.rs:73]: Yeah, I only really wrote `active_mut` for debugging. Accessing the hellevator should be gated behind `status()` which provides more detailed info for the different stages. 

I suppose you mainly readded this to remove the old reward state, but for that a `clear_rewards()` function on`HellevatorEvent` would make more sense.

File: src/gamestate/unlockables.rs [hunk 2]
```diff
@@ -102,6 +101,10 @@ pub struct Hellevator {
     pub next_reset: Option<DateTime<Local>>,
     pub start_contrib_date: Option<DateTime<Local>>,
 
+    /// This field is only updated during login. After claiming the reward
+    /// using `Command::HellevatorClaimDailyYesterday`, you must manually
+    /// reset this field.
+    pub rewards_yesterday: Option<HellevatorDailyReward>,
```
Comment (the-marenga) [inline src/gamestate/unlockables.rs:107]: Have you checked if calling poll (`Command::Update`) refreshes this?

**Review by the-marenga — CHANGES_REQUESTED** [review]

**Comment by the-marenga** [comment]
I guess this got turned into a draft whilst I was lookin at it, lol

**Comment by mzimnn** [comment]
Yeah, now that some of my characters can claim a yesterday's daily reward, I tried to see if I can find out more. Just in case, I turned it into a draft. Actually, I may have found a way to update the game state.

Here's what I found out:

In the login response you receive information about yesterday's daily rewards. I couldn't find any other way to access this information, such as by polling, etc.

Example:
```
gtdailyrewardyesterday:15/18/1/1/2/2315004/20700/6660/1712/2568/2/1/1500/650000
```

After sending the command `GroupTournamentClaimDailyYesterday:`, the server responds with:

```
gtdailyrewardclaimed:15/18/1/1/2/2315004/20700/6660/1712/2568/
```

The first 10 fields in `gtdailyrewardclaimed` are always the same as in `gtdailyrewardyesterday`. However, the same response is also sent if you claim today's daily rewards using the command `GroupTournamentClaimDaily:`.

I think it could be possible to check if the fields of `gtdailyrewardclaimed` match either `Hellevator.rewards_today` or `Hellevator.rewards_yesterday` and then reset the respective field in the `Hellevator` struct. However, if both today's and yesterday's daily rewards have the same values, this logic fails – not sure how likely this is.

But I guess this is still a better solution than manually updating the game state. What do you think?

**Comment by the-marenga** [comment]
> think it could be possible to check if the fields of gtdailyrewardclaimed match either Hellevator.rewards_today or Hellevator.rewards_yesterday and then reset the respective field in the Hellevator struct.
> But I guess this is still a better solution than manually updating the game state. What do you think?
> EDIT: I think this is even less of an issue since Hellevator.rewards_today is properly updated

Yeah, that sounds a bit hacky, but it is still a lot better than manually having to update it. I aggree completely

File: src/gamestate/unlockables.rs (hunk 1 above)
Comment (mzimnn) [reply src/gamestate/unlockables.rs]: With the latest changes this is not relevant anymore.

File: src/gamestate/mod.rs [hunk 3]
```diff
@@ -1325,12 +1325,45 @@ impl GameState {
...
+                            if rewards_yesterday == rewards_claimed {
+                                // The game doesn't update this value itself, so
+                                // we do it manually.
+                                //
+                                // NOTE: In practice this should be good enough,
+                                // but theoretically it is possible that both
+                                // kind of rewards have the same value. In this
+                                // case, it could happen that yesterday's daily
+                                // reward is reset even though today's daily
+                                // reward was claimed.
```
Comment (the-marenga) [inline src/gamestate/mod.rs:1362]: Just a though. If yesterdays response consists of only "gtdailyrewardclaimed", whilst todays claim also contains "gtdailyreward" (or whatever that sends to update the value), could we not also check if `new_vals.len() == 1` or `!new_vals.contains_key("gtdailyreward")` here?.  

In addition, have you ever looked at `sub_key()` on ResponseVal? That contains nothing 99.9% of the times, but this here could be one of those cases, where they add some metadata to differentiate the two cases.

**Review by the-marenga — APPROVED** [review]
Looks good! I just had a though about the parsing again, but nothing that would actually block merging this, if you think those are dead ends

File: src/gamestate/mod.rs (hunk 3 above)
Comment (mzimnn) [reply src/gamestate/mod.rs:1362]: This is actually a good idea. I'll test if this works as well, then we can get rid of this edge case.

Comment (mzimnn) [reply src/gamestate/mod.rs:1362]: See: https://github.com/the-marenga/sf-api/pull/162
